cmdstanpy/model.py

```diff
@@ -652,91 +677,130 @@ def _read_progress(self, proc: subprocess.Popen, pbar: List[Any]) -> bytes:
         # gather stdout
         stdout = b''
 
-        # iterate while process is sampling
-        # process stdout line by line
-        while proc.poll() is None:
-            # read line
-            output = proc.stdout.readline()
-            # gather stdout
-            stdout += output
-            # decode bytes to string
-            output = output.decode('utf-8').strip()
-
-            # force refresh warmup_pbar after warmup has ended
-            refresh_warmup = True
-            # process Iteration output
-            if output.startswith('Iteration'):
-                # run regular expression
-                match = re.search(pattern_compiled, output)
-                if match:
-                    # raw_count = warmup + sampling
-                    raw_count = int(match.group(1))
-                    if match.group(2).lower() == 'warmup':
-                        count, count_warmup = (
-                            # steps
-                            raw_count - count_warmup,
-                            # cumulative count
-                            raw_count,
-                        )
-                        # increment progressbar by 'count'
-                        pbar_warmup.update(count)
-                    elif match.group(2).lower() == 'sampling':
-                        # refresh warmup and close the progress bar if
-                        # all the warmup samples are seen
-                        if refresh_warmup:
-                            pbar_warmup.refresh()
-                            if count_warmup == num_warmup:
+        try:
+            # iterate while process is sampling
+            # process stdout line by line
+            while proc.poll() is None:
+                # read line
```

Comment:
most of the line-by-line commentary is unnecessary - comment on line #681 is sufficient.
should be removed.

Reply:
completely understand - I do this too.
merged this yesterday - we can do the cleanup via another PR.